features/vaadin-dashlets/dashlet-bsm/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/BSMDashlet.java

```diff
@@ -0,0 +1,220 @@
+        super(name, dashletSpec);
+        /**
+         * Setting the member fields
+         */
+        m_businessServiceManager = transactionAwareBeanProxyFactory.createProxy(businessServiceManager);
+
+        /**
+         * Retrieve the config...
+         */
+
+        m_businessServiceSearchCriteria = BSMConfigHelper.fromMap(getDashletSpec().getParameters());
+    }
+
+    @Override
+    public DashletComponent getWallboardComponent() {
+        if (m_wallboardComponent == null) {
+            m_wallboardComponent = new AbstractDashletComponent() {
+                private GridLayout m_gridLayout = new GridLayout(10, 1);
+
+                {
+                    m_gridLayout.setCaption(getName());
+                    m_gridLayout.setWidth("100%");
+                    refresh();
+                }
+
+                private void injectStyles() {
+                    Page.getCurrent().getStyles().add(".severity { color:#000000; padding-left: 20px; background-position: top left; background-repeat: repeat-y; }");
+                    Page.getCurrent().getStyles().add(".severity.Critical{ color:#000000; background-color: #F5CDCD; border-top: 2px solid #CC0000 !important; background-image: url(images/bgCritical.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Major{ color:#000000; background-color: #FFD7CD; border-top: 2px solid #FF3300 !important; background-image: url(images/bgMajor.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Minor{ color:#000000; background-color: #FFEBCD; border-top: 2px solid #FF9900 !important; background-image: url(images/bgMinor.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Warning{ color:#000000; background-color: #FFF5CD; border-top: 2px solid #FFCC00 !important; background-image: url(images/bgWarning.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Indeterminate{ color:#000000; background-color: #EBEBCD; border-top: 2px solid #999000 !important; background-image: url(images/bgIndeterminate.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Normal{ color:#000000; background-color: #D7E1CD; border-top: 2px solid #336600 !important; background-image: url(images/bgNormal.png); }");
+                    Page.getCurrent().getStyles().add(".severity.Cleared { color:#000000; background-color: #EEEEEE; border-top: 2px solid #999999 !important; background-image: url(images/bgCleared.png); }");
+                    Page.getCurrent().getStyles().add(".severity.bright { color:#000000; padding-left: 20px; background-position: top left; background-repeat: repeat-y; }");
+                    Page.getCurrent().getStyles().add(".severity.nobright { color:#000000; background-image: none;}");
+                }
+
+                @Override
+                public void refresh() {
+                    m_gridLayout.removeAllComponents();
+                    injectStyles();
```

Comment:
The same styles get injected on the page every time the dashlet is refreshed. Since these are never removed, the DOM just keeps growing with copies of the same styles every refresh.

(Many of the other existing dashlets have this same problem.)